docs/articles/lavalink/implementation.md

```diff
@@ -0,0 +1,127 @@
+```csharp
+using DSharpPlus.Net;
+using DSharpPlus.Lavalink;
+```
+Next, make sure you have a node connection property in your class. You will need this later on for commands.
+```csharp
+public static LavalinkNodeConnection LavalinkNode;
+```
+
+After that, we will need to create a configuration for our extension to use. This is where the special values from the server configuration are used.
+```csharp
+var restConfig = new ConnectionEndpoint 
+{
+    Hostname = "localhost", //From your server configuration.
+    Port = 8080 //From your server configuration
+};
+
+var socketConfig = new ConnectionEndpoint
+{
+    Hostname = "localhost", //From your server configuration.
+    Port = 8080 //From your server configuration
+};
+
+var lcfg = new LavalinkConfiguration
+{
+    Password = "youshallnotpass", //From your server configuration.
+    RestEndpoint = restConfig,
+    SocketEndpoint = socketConfig
+};
+```
+Finally, initialize the extension.
+```csharp
+Lavalink = discord.UseLavalink(lcfg);
+```
+
+## Connecting with Lavalink
+
+We are now ready to connect to the server. Call the Lavalink extension's connect method and pass the configuration. Make sure to call this **after** your Discord client connects. Assign the connection to your `LavalinkNode`.
+
+```csharp
+LavalinkNode = await Lavalink.ConnectAsync(lcfg);
+```
+
+Your main bot file should now look like this: 
+
+```csharp
+using System.Threading.Tasks;
+using DSharpPlus;
+using DSharpPlus.Net;
+using DSharpPlus.Lavalink;
+
+namespace MyFirstMusicBot
+{
+    class Program
+    {
+        public static DiscordClient Discord;
+        public static LavalinkNodeConnection LavalinkNode;
+
+        static void Main(string[] args)
+        {
+            MainAsync(args).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        static async Task MainAsync(string[] args)
+        {
+            Discord = new DiscordClient(new DiscordConfiguration
+            {
+                Token = "<token_here>",
+                TokenType = TokenType.Bot,
+                UseInternalLogHandler = true,
+                LogLevel = LogLevel.Debug
+            });
+
+            var restConfig = new ConnectionEndpoint
+            {
+                Hostname = "localhost", //From your server configuration.
+                Port = 8080 //From your server configuration
+            };
+
+            var socketConfig = new ConnectionEndpoint
+            {
+                Hostname = "localhost", //From your server configuration.
+                Port = 8080 //From your server configuration
+            };
+
+            var lcfg = new LavalinkConfiguration
+            {
+                Password = "youshallnotpass", //From your server configuration.
+                RestEndpoint = restConfig,
+                SocketEndpoint = socketConfig
+            };
+
+            Discord.UseLavalink();
```

Comment:
You make a config and then ignore it.

Reply:
The config is used [here](https://github.com/DSharpPlus/DSharpPlus/pull/603/files#diff-3f1755fce1199bf8ef4059555821e17bR101):

```csharp
LavalinkNode = await Lavalink.ConnectAsync(lcfg); //Make sure this is after Discord.ConnectAsync().
```